pypicloud/cache/sql.py

```diff
@@ -163,6 +164,62 @@ def distinct(self):
             .order_by(SQLPackage.name).all()
         return [n[0] for n in names]
 
+    def search(self, criteria, query_type):
+        """
+        Perform a search.
+
+        Queries are performed as follows:
+
+            For the AND query_type, queries within a column will utilize the
+            AND operator, but will not conflict with queries in another column.
+
+                (column1 LIKE '%a%' AND column1 LIKE '%b%')
+                OR
+                (column2 LIKE '%c%' AND column2 LIKE '%d%')
+
+            For the OR query_type, all queries will utilize the OR operator:
+
+                (column1 LIKE '%a%' OR column1 LIKE '%b%')
+                OR
+                (column2 LIKE '%c%' OR column2 LIKE '%d%')
+
+        """
+        conditions = []
+        packages = []
+        for key, queries in criteria.items():
+            # Make sure search key exists in the package class.
+            # It should be either "name" or "summary".
+            field = getattr(self.package_class, key, None)
+            if not field:
+                continue
+
+            column_conditions = []
+
+            for query in queries:
+                # Generate condition and add to list
+                condition = field.like('%' + query + '%')
+                column_conditions.append(condition)
+
+            # Check if conditions for this column were generated
+            if column_conditions:
+                if query_type == 'and':
+                    operator = and_
+                else:
+                    operator = or_
+                conditions.append(operator(*column_conditions))
```

Comment:
I reworked the logic here to match the logic that the other cache backends used. I explained it in the docstring.

Let me know if this makes sense to you.